compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/codegen/typ.rs

```diff
@@ -374,7 +374,22 @@ impl<'tcx> GotocCtx<'tcx> {
         // code.  See the implementation of pretty_print_region on line 1720 in
         // compiler/rustc_middle/src/ty/print/pretty.rs.
         let name = name.replace(" + \'static", "").replace("\'static ", "");
-        name
+
+        // Crate resolution: mangled names need to be distinct across different versions
+        // of the same crate that could be pulled in by dependencies. However, RMC's
+        // treatment of FFI C calls asssumes that we generate the same name for structs
+        // as the C name, so don't mangle in that case.
```

Comment:
This is a hack. We should note it as such, and have a tracking issue to do the right thing here.

Reply:
Filed and adding